<a href="https://colab.research.google.com/github/dscoool/chungbuk/blob/main/tensorflow_beginner.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 파이썬 텐서플로우 시작하기!!

간단한 예제(tensorflow)를 아무 생각 없이 따라 돌려 봅시다.

tensorflow라는 모듈을 사용할게요!!


* 예제 설명: https://guide.ncloud-docs.com/beta/docs/tensorflow-for-beginner


먼저 프로그램에 텐서플로 라이브러리를 임포트합니다


In [ ]:
import tensorflow as tf


[MNIST 데이터셋](https://brunch.co.kr/@minwoo/25) 을 로드하여 준비합니다. 샘플 값을 정수에서 실수(float, 부동소수)로 변환합니다:

In [ ]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11490434/11490434 [==============================] - 0s 0us/step


In [ ]:
# 데이터가 궁금하면 한 개씩 주석을 풀고 실행해 봅시다!!
# 어떤 데이터가 로드되었을까요?
# mnist.load_data()
# x_train
# x_test
# y_train
y_test

array([7, 2, 1, ..., 4, 5, 6], dtype=uint8)

층을 차례대로 쌓아 tf.keras.Sequential 모델을 만듭니다. 훈련에 사용할 옵티마이저(optimizer)와 손실 함수를 선택합니다:

In [ ]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

모델을 훈련하고 평가합니다:

In [ ]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test,  y_test, verbose=2)


''' 예상 결과
Train on 60000 samples
Epoch 1/5
WARNING:tensorflow:Entity <function Function._initialize_uninitialized_variables.<locals>.initialize_variables at 0x7f2d11707048> could not be transformed and will be executed as-is. Please report this to the AutoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: module 'gast' has no attribute 'Num'
WARNING: Entity <function Function._initialize_uninitialized_variables.<locals>.initialize_variables at 0x7f2d11707048> could not be transformed and will be executed as-is. Please report this to the AutoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: module 'gast' has no attribute 'Num'
60000/60000 [==============================] - 4s 68us/sample - loss: 0.2941 - accuracy: 0.9140
Epoch 2/5
60000/60000 [==============================] - 4s 62us/sample - loss: 0.1396 - accuracy: 0.9587
Epoch 3/5
60000/60000 [==============================] - 4s 62us/sample - loss: 0.1046 - accuracy: 0.9680
Epoch 4/5
60000/60000 [==============================] - 4s 62us/sample - loss: 0.0859 - accuracy: 0.9742
Epoch 5/5
60000/60000 [==============================] - 4s 62us/sample - loss: 0.0724 - accuracy: 0.9771
10000/1 - 0s - loss: 0.0345 - accuracy: 0.9788
[0.06729823819857557, 0.9788]
'''

Epoch 1/5
1875/1875 [==============================] - 15s 8ms/step - loss: 0.2879 - accuracy: 0.9163
Epoch 2/5
1875/1875 [==============================] - 18s 9ms/step - loss: 0.1413 - accuracy: 0.9579
Epoch 3/5
1875/1875 [==============================] - 15s 8ms/step - loss: 0.1053 - accuracy: 0.9687
Epoch 4/5
1875/1875 [==============================] - 17s 9ms/step - loss: 0.0858 - accuracy: 0.9734
Epoch 5/5
1875/1875 [==============================] - 18s 9ms/step - loss: 0.0745 - accuracy: 0.9762
313/313 - 1s - loss: 0.0730 - accuracy: 0.9780 - 661ms/epoch - 2ms/step


[0.07300049066543579, 0.9779999852180481]

훈련된 이미지 분류기는 이 데이터셋에서 약 98%의 정확도를 달성합니다. 더 자세한 내용은 TensorFlow 튜토리얼 을 참고해 주십시오.

* 텐서플로우 튜토리얼- https://www.tensorflow.org/tutorials

```
# 코드로 형식 지정됨
```

